src/Symfony/Component/Routing/Generator/UrlGenerator.php

```diff
@@ -294,6 +294,18 @@ protected function doGenerate($variables, $defaults, $requirements, $tokens, $pa
             return $a == $b ? 0 : 1;
         });
 
+        // force string for an object. See https://bugs.php.net/bug.php?id=66966
+        foreach ($extra as $paramName => $param) {
+            if (!\is_object($param)) {
+                continue;
+            }
+            if (!\is_callable([$param, '__toString'])) {
+                $message = 'Error when generating an URL for "{route}". Cannot convert object of type {type} to string for parameter "{parameter}".';
+                throw new InvalidParameterException(strtr($message, ['{route}' => $name, '{type}' => get_debug_type($param), '{parameter}' => $paramName]));
```

Comment:
Why not use `sprintf()`, we use that everywhere else, dont we?

```suggestion
                throw new InvalidParameterException(\sprintf('Error when generating an URL for "%s". Cannot convert object of type "%s" to string for parameter "%s".', $name, get_debug_type($param), $paramName));
```

Reply:
Indeed, the same file is using `sprintf` 2 times and `strtr` 2 times

Reply:
I quick (and maybe flawed) regex search showed that we are only using `strtr` when throwing an exception twice in this repo... both in this file =)

Let's use `sprintf`.